magento/pwa-studio
# Add pjpg format only if image type of not png.

## Description

This PR adds logic in `makeUrl.js` file in `venia-ui` to handle png image requests.

Currently, the `makeOptimizedUrl` function does not consider the file type while adding URLSearchParams. It adds `webp` and `pjpg` in that respective order. This means if the server can not handle `webp` it will consider `pjpg`. `webp` is compatible with `png`, so when the user requests a `png` it will be returned with a transparent background.

![image](https://user-images.githubusercontent.com/35203638/81865380-75c5f600-9533-11ea-9277-3a1afc6e2a56.png)

If the server does not support `webp` since the current code does not regard `png` and applies `pjpg` as the format, instead of a transparent background, we will get the image with a black background.

![image](https://user-images.githubusercontent.com/35203638/81865689-d2291580-9533-11ea-9f60-5c9bd5d3b604.png)

But with the current change, instead of `pjpg` the UI will request a `png` instead.

![image](https://user-images.githubusercontent.com/35203638/81865862-07cdfe80-9534-11ea-9e4c-8e79fc1c6698.png)

All this happens only if the backend does not support `webp` and is served by a backend that does not handle `pjpg` properly. For instance, we were not able to reproduce this issue till we switched off fastly, because fastly handles `pjpg` well and does not return a black background. When we switched off fastly, the backend image optimization was handled by hastily which was retuning the black background.

I am not sure if the approach taken by fastly or hastily is correct but one thing is for sure, we have to send format as `png` for a `png` image request irrespective of the backend.

## Troubleshooting Tips:
1. Check if the backend service supports `webp`. If it does, everything should be fine.
2. If it does not, check the file format requested in the `URLSearchParams`.

## Related Issue
Closes #2340 

### Verification Stakeholders
@zetlen 
@jimbo 
@Jordaneisenburger 

### Verification Steps

To test use this, use this backend URL:
`MAGENTO_BACKEND_URL=https://staging-5em2ouy-mfwmkrjfqvbjk.us-4.magentosite.cloud/`

1. Go to https://magento-venia-concept-7bnnn.local.pwadev:8914/simple13.html
2. With fastly enabled/disabled everything should work fine. You should see a transparent background when you open the image in the network tab.

## Screenshots / Screen Captures (if appropriate)

## Checklist
None
